frontend/src/components/SignUp/SignUp.js

```diff
@@ -16,41 +22,53 @@ class SignUp extends Component {
     }));
   }
 
+  handleChange = (event) => {
+    this.setState({ [event.target.name] : event.target.value});
+  }
+
+  handleSubmit = (event) => {
+    //api call goes here
+    event.preventDefault()
+    createUser(this.state).then(response => {
+      console.log(response)}).catch(error => {console.log(error.response.data)})
+  }
+
+
   render() {
     return (
       <div className="leftLoginCard loginCard">
         <h3 className="loginHeader">Create an Account</h3>
-        <form className="form-signInUp">
+        <form className="form-signInUp" onSubmit={this.props.mimicLogin}>
           <div className="form-group row">
               <div className="name col">
                 <div className="name row">
-                  <input type="text" id="inputFirstName" className="sign-in-input firstName col-md form-control-lg" required="" autoFocus="" placeholder="First" onFocus={(event) => (event.target.setAttribute("placeholder", ""))}
-onBlur={(event) => (event.target.setAttribute("placeholder", "First"))}/>
+                  <input type="text" id="inputFirstName" name="firstName" className="sign-in-input firstName col-md form-control-lg" placeholder="First" onFocus={(event) => (event.target.setAttribute("placeholder", ""))}
+onBlur={(event) => (event.target.setAttribute("placeholder", "First"))} onChange={(event) => this.handleChange(event)}/>
```

Comment:
I think you can just `onChange=this.handleChange`. I wonder if there's a way we can abstract out this `onFocus`/`onBlur`/`onChange` stuff into a repeatable pattern (like a small component or something) so we don't have to keep repeating it over and over.

Reply:
I agree. Maybe in a separate ticket?  I want to try and complete the login functionality first. . . did change to `this.handleChange`. Didn't even know I could do that!